docker/src/main/java/org/jclouds/docker/compute/strategy/DockerComputeServiceAdapter.java

```diff
@@ -136,6 +140,14 @@ public DockerComputeServiceAdapter(DockerApi api) {
               .publishAllPorts(true)
               .privileged(true);
 
+      if (templateOptions.getDirectPorts().isPresent()) {
```

Comment:
Agree. Could have a `Map<Integer, Integer>` type here, instead? This would be mappings of host port to container port (this is the order  Docker uses with `--publish`.)